### PR TITLE
add k8s owner

### PR DESCRIPTION
**Pull request by eranra**

### CODE REVIEW

**Comment by eranra** [comment]
image

Deamonset owner

**Comment by eranra** [comment]
image

stateful owner

**Comment by eranra** [comment]
More types ::

image

**Comment by eranra** [comment]
@jotak if you can review ... I think that this is what we need for owner

**Review by jotak — APPROVED** [review]
thanks, looks good to me!

File: pkg/pipeline/transform/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -189,7 +244,7 @@ func (k KubeData) InitFromConfig(kubeConfigPath string) error {
 	return nil
 }
 
-func (k KubeData) initInformers(client kubernetes.Interface) error {
+func (k *KubeData) initInformers(client kubernetes.Interface) error {
 	//defer close(stopChan)
```
Comment (KalmanMeth) [inline pkg/pipeline/transform/kubernetes/kubernetes.go:248]: Is this line no longer relevant? Delete.

**Comment by jotak** [comment]
actually just a small remark, it seems like `Type` is all lower case whereas OwnerType is CamelCase, would it be easy to also have CamelCase for Pod / Service / Node (so that it matches the k8s kinds) ?

File: pkg/pipeline/transform/kubernetes/kubernetes_test.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+type InformerInterface interface {
+	cache.SharedInformer
+	AddIndexers(indexers cache.Indexers) error
+	GetIndexer() cache.Indexer
+}
+
+func (indexMock *IndexerMock) ByIndex(indexName, indexedValue string) ([]interface{}, error) {
+	//indexMock.Mock.Called(indexName, indexedValue)
```
Comment (KalmanMeth) [inline pkg/pipeline/transform/kubernetes/kubernetes_test.go:46]: Delete unused function call?

**Review by KalmanMeth — APPROVED** [review]

File: pkg/pipeline/transform/kubernetes/kubernetes.go (hunk 1 above)
Comment (eranra) [reply pkg/pipeline/transform/kubernetes/kubernetes.go:248]: removed, done

File: pkg/pipeline/transform/kubernetes/kubernetes_test.go (hunk 2 above)
Comment (eranra) [reply pkg/pipeline/transform/kubernetes/kubernetes_test.go:46]: removed, done

File: pkg/pipeline/transform/kubernetes/kubernetes.go (hunk 1 above)
Comment (eranra) [reply pkg/pipeline/transform/kubernetes/kubernetes.go:248]: done

**Comment by eranra** [comment]
> actually just a small remark, it seems like `Type` is all lower case whereas OwnerType is CamelCase, would it be easy to also have CamelCase for Pod / Service / Node (so that it matches the k8s kinds) ?

I changed to CamelCase to match :-)

**Pull request merged**